concordium-consensus/tests/consensus/ConcordiumTests/KonsensusV1/Types.hs

```diff
@@ -34,28 +34,38 @@ genBlsSignature = flip Bls.sign someBlsSecretKey . BS.pack <$> vector 10
 genQuorumSignature :: Gen QuorumSignature
 genQuorumSignature = QuorumSignature <$> genBlsSignature
 
+-- |Generate a block hash for a genesis
+genGenesisBlockHash :: Gen BlockHash
+genGenesisBlockHash = BlockHash . Hash.Hash . FBS.pack <$> vector 32
+
 -- |Generate a quorum certificate in a way that is suitable for testing serialization.
-genQuorumCertificate :: Gen QuorumCertificate
-genQuorumCertificate = do
+-- The genesis block hash must be provided.
```

Comment:
Can you elaborate on the Maybe then? It is unclear what "must" means when you can omit it.